File: 1.两数之和.cpp
class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        unordered_map<int, int> map;
        for (size_t i = 0; i < nums.size(); i++)
        {
            int complement = target - nums[i];
            if (map.find(complement) != map.end())
            {
                int result[] = {map[complement], i};
                return vector<int>(result, result + 2);
            }
            map[nums[i]] = i;
        }
        return vector<int>{};
    }
};

File: 1.两数之和.py
#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashmap = {}
        for index, value in enumerate(nums):
            complement = target - value
            if complement in hashmap:
                return {hashmap[complement],index}
            hashmap[value] = index
